galsim/scene.py

```diff
@@ -194,47 +271,104 @@ def __init__(self, file_name=None, image_dir=None, dir=None, preload=False, nois
         self.orig_index = np.arange(len(self.param_cat))
         mask = np.ones(len(self.orig_index), dtype=bool)
 
-        # If requested, select galaxies based on existence of a usable fit.
-        if exclude_fail:
+        # If requested, select galaxies in a way that excludes suspect postage stamps (e.g., with
+        # deblending issues), suspect parametric model fits, or both of the above plus marginal
+        # ones.
+        if exclusion_level == 'bad_ps' or exclusion_level == 'marginal':
+            # This 'exclusion_level' involves placing cuts on the S/N of the object detection in the
+            # original postage stamp, and on issues with masking that can indicate deblending or
+            # detection failures.  These cuts were used in GREAT3.
+            # In the case of the masking cut, in some cases there are messed up ones that have a 0
+            # for self.selection_cat['peak_image_pixel_count'].  To make sure we don't divide by
+            # zero (generating a RuntimeWarning), and still eliminate those, we will first set that
+            # column to 1.e-5.
+            # We choose a sample-dependent mask ratio cut, since this depends on the peak object
+            # flux, which will differ for the two samples (and we can't really cut on this for
+            # arbitrary user-defined samples).
+            if use_sample == "23.5":
+                cut_ratio = 0.2
+                sn_limit = 20.0
+            else:
+                cut_ratio = 0.8
+                sn_limit = 12.0
+            if self.selection_cat is not None:
+                div_val = self.selection_cat['peak_image_pixel_count']
+                div_val[div_val == 0.] = 1.e-5
+                mask &= ( (self.selection_cat['sn_ellip_gauss'] >= sn_limit) &
+                          ((self.selection_cat['min_mask_dist_pixels'] > 11.0) |
+                           (self.selection_cat['average_mask_adjacent_pixel_count'] / \
+                               div_val < cut_ratio)) )
+
+        if exclusion_level == 'bad_fits' or exclusion_level == 'marginal':
```

Comment:
Personally, I like the earlier style: `if exclusion_level in ['bad_fits', 'marginal']`.